packages/o11ytracesdb/src/query.ts

```
@@ -0,0 +1,768 @@
+        case "gt": return a > b;
+        case "gte": return a >= b;
+        case "lt": return a < b;
+        case "lte": return a <= b;
+      }
+      break;
+    }
+
+    case "regex": {
+      if (typeof attrVal !== "string" || typeof pred.value !== "string") return false;
+      try {
+        const re = pred._compiledRegex ?? (pred._compiledRegex = new RegExp(pred.value));
```

Biome (2.4.13)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)